src/github_actions_exporter.py

```diff
@@ -2,32 +2,58 @@
 # Copyright 2023 Canonical Ltd.
 # See LICENSE file for licensing details.
 
-"""Helper class used to manage interactions with GitHub Actions Exporter."""
+"""Helper module used to manage interactions with GitHub Actions Exporter."""
 
 from re import findall
+from typing import Dict
 
+from ops.model import Container
 
-# There is only one method for this class.
-class GitHubActionsExporter:  # pylint: disable=too-few-public-methods
-    """This class handles interactions with GitHub Actions Exporter."""
-
-    def __init__(
-        self,
-        charm,
-    ):
-        """Construct."""
-        self._charm = charm
-
-    def get_exporter_version(self) -> str:
-        """Retrieve the current version of GitHub Actions Exporter.
-
-        Returns:
-            The  GitHub Actions Exporter version installed.
-        """
-        container = self._charm.unit.get_container("github-actions-exporter")
-        process = container.exec(
-            ["/srv/gh_exporter/github-actions-exporter", "--version"], user="gh_exporter"
-        )
-        version_string, _ = process.wait_output()
-        version = findall("[0-9a-f]{5,40}", version_string)
-        return version[0][0:7] if version else ""
+from charm_state import CharmState
+
+
+def exporter_environment(state: CharmState) -> Dict[str, str]:
+    """Generate a environment dictionary from the charm configurations.
+
+    Args:
+        state: The state of the charm.
+
+    Returns:
+        A dictionary representing the GitHub Actions Exporter environment variables.
+    """
+    return {
+        "GITHUB_WEBHOOK_TOKEN": f"{state.github_webhook_token}",
+        "GITHUB_API_TOKEN": f"{state.github_api_token}",
+        "GITHUB_ORG": f"{state.github_org}",
+    }
+
+
+def exporter_version(container: Container, state: CharmState) -> str:
+    """Retrieve the current version of GitHub Actions Exporter.
+
+    Args:
+        container: The container of the charm.
+        state: The state of the charm.
+
+    Returns:
+        The  GitHub Actions Exporter version installed.
+    """
+    process = container.exec([state.github_exporter_command, "--version"], user="gh_exporter")
+    version_string, _ = process.wait_output()
+    version = findall("[0-9a-f]{5,40}", version_string)
+    return version[0][0:7] if version else ""
```

Comment:
Should this return `None` if it can't find the version?

Reply:
Good point for discussion! My preference is to keep it as an empty string, as it is falsey and you can do a check on the returned value just fine. This (in my mind) is consistent with what `findall` is also doing - return a list of values or an empty list (which is also falsey and you do your checks over it if needed).